classes/various/Matrix.sc

```diff
@@ -10,15 +10,24 @@ Matrix[slot] : Array {
 	*newClear { arg rows=1, cols=1; // return (rows x cols) - zero matrix
 		^super.fill(rows, { Array.newClear(cols).fill(0) });
 	}
-	*with { arg array; // return matrix from 2D array (array of rows)
-		var rows;
-		if((array.flat == array.flop.flop.flat)
-		.and(array.flatten.every({arg element; element.isNumber})) ,{
+	/* Michael McCrea, 2019 */
+	// refactor to avoid .flop.flop, which if called on Array, will expose
+	// it to a bug that could fail on large arrays.
+	// also, this is more efficient
+	*with { |array| // return matrix from 2D array (array of rows)
+		var shapes, shapeTest, numTest, rows;
+
+		shapes = (array.asArray).collect(_.shape).flatten;
+
+		shapeTest = shapes.every(_ == shapes[0]);
+		numTest = array.flatten.every(_.isNumber);
+
+		if((shapeTest and: numTest), {
```

Comment:
it would be more efficient to write:
```
shapeTest = { shapes.every(_ == shapes[0]) };
numTest = { array.flatten.every(_.isNumber) };
f((shapeTest and: numTest), {
```